Collected small changes and fixes

akohlmey opened the pull request:
**Summary**

This pull request combines multiple small changes and fixes.

**Related Issue(s)**

Fixes #3499 
Fixes #3496 
Fixes #3484 
Fixes #3476 
Supersedes and thus closes #3483 
Supersedes and thus closes #3493 

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Some regression tests will fail because of the different reneighboring steps causing differences in forces and thus diverging trajectories.

**Implementation Notes**

The following individual changes are inlcuded:
- change the default setting for neighbor lists to reduce delay=10 to delay=0, add note to documentation
- delete *atomfile* style variables with the `clear` command to avoid segmentation faults on subsequent `read_data` calls
- multiple small fixes for issues related to PR #3478 
- correctly detect the `clang-format` version on Ubuntu 22.04LTS with CMake (see issue #3484)
- update list and CMake script code for installing include files from the main source folder with "make install"
- update PLUMED package library to version 2.8.1 and also support 2.7.5
- small FreeBSD portability enhancement
- report the new communication cutoff in the warning when it is extended for some of the GPU pair styles
- increase the communication cutoff for TIP4P pair styles to avoid errors (adapted from PR #3483)
- corrects column selection limits in thermo_modify format (see issue #3476)
- provide access to `lammps_config_has_package()` in the python module
- stop with an error when trying to use `-D ENABLE_TESTING=on` with too old compilers (e.g. GNU 4.8.x)
- skip `FortranOpen` tests if no suitable F90 MPI module is found.
- skip some unit tests that require the MOLECULE package in case it is not available
- improve a few error messages
- more detailed message with the replicate command (alternative to PR #3493)
- reformat sources with clang-format
- implement workaround resulting in proper hyperlinks to Fortran functions in the manual (addresses issue #3496)

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

akohlmey committed:
fix typo

akohlmey committed:
use for scoped loop variables consistently

akohlmey committed:
silence compiler warning

akohlmey committed:
make lammps_config_has_package() accessible in Python module

akohlmey committed:
we cannot test the Pizza.py dump class without the MOLECULE package

akohlmey committed:
need to skip tests if MOLECULE package is missing

akohlmey committed:
flag lammps.has_package() python method as new

akohlmey committed:
improve error messages

akohlmey committed:
cosmetic

akohlmey committed:
fix bug in thermo_modify format column check

akohlmey committed:
more false positives

akohlmey committed:
improve warning messages

akohlmey committed:
detect clang-format version on Ubuntu 22.04LTS

akohlmey committed:
include clang-format

akohlmey committed:
Increase communication cutoff for TIP4P pair styles, if needed

This avoids error of H atom not found when the O atom is a ghost.

akohlmey committed:
change warning message and get the comm cutoff properly

akohlmey committed:
update install targets for main folder headers for "make install"

evoyiatzis commented:
In doc/src/run_style.rst line 104 there is a spelling error "its onn optimal layout" instead of "its own optimal layout"

akohlmey committed:
fix typo and remove correspondign false positive

akohlmey commented:
> In doc/src/run_style.rst line 104 there is a spelling error "its onn optimal layout" instead of "its own optimal layout"

Thanks. I found that this passed the spell check because of an incorrect entry in the false positives list.

akohlmey committed:
put a few more details on restrictions when parsing types early on.

akohlmey committed:
also print atom-step/s performance metric

akohlmey committed:
support Gatom-step/s as well

akohlmey committed:
change neighbor list defaults to delay=0, every=1, check=yes

akohlmey committed:
cosmetic

akohlmey committed:
update remarks on neighbor list settings

akohlmey committed:
apply clang-format

sjplimp commented on doc/src/neigh_modify.rst:
```diff
@@ -71,30 +71,52 @@ Description
+   the (parallel) performance of LAMMPS and the correctness of the
+   simulation results.  Since building the neighbor lists is time
+   consuming, doing it less frequently can speed up a calculation.  If
+   the lists are rebuilt too infrequent, however, interacting pairs are
```
infrequently

sjplimp commented on doc/src/neigh_modify.rst:
```diff
@@ -71,30 +71,52 @@ Description
+   simulation results.  Since building the neighbor lists is time
+   consuming, doing it less frequently can speed up a calculation.  If
+   the lists are rebuilt too infrequent, however, interacting pairs are
+   missed and thus the resulting pairwise interactions incorrect.  The
```
interacting pairs can potentially be missed

sjplimp commented on doc/src/neigh_modify.rst:
```diff
@@ -71,30 +71,52 @@ Description
+   simulated system (density, geometry, topology, temperature,
+   pressure), the force field parameters and settings, the size of the
+   timestep, neighbor list skin distance and more.  The default settings
+   are chosen to be very conservative to promote correctness of the
```
to guarantee correctness

sjplimp commented on doc/src/neigh_modify.rst:
```diff
@@ -71,30 +71,52 @@ Description
+   are chosen to be very conservative to promote correctness of the
+   simulation.  They depend on the *check* flag heuristics to reduce the
+   number of neighbor list rebuilds at a minor expense for executing the
+   check.  Determining the correctness of a specific coice of neighbor
```
choice

sjplimp reviewed (changes requested):
a few typos & rewordings

akohlmey committed:
update wording and fix typo

akohlmey commented:
> a few typos & rewordings

changes are applied

akohlmey committed:
KOKKOS_ARCH output is always empty, so no need to print it

akohlmey committed:
address portability issues to the flang LLVM Fortran 2018 compiler

akohlmey committed:
simplify

akohlmey committed:
fix typo

akohlmey committed:
add CMake check that will refuse compilation of unit tests or skip tests

This is mainly because the default compilers on RHEL/CentOS 7.x are
not sufficient to compile googletest. Also some Fortran module test
requires a working F90 module and others are more recent Fortran compiler.

akohlmey committed:
update/clarify and make more consistent

akohlmey committed:
more strictly follow the Fortran 2003 standard requirements

akohlmey committed:
update dipole pair style examples and docs

akohlmey committed:
Merge branch 'develop' into collected-small-changes

akohlmey committed:
delete atomfile style variables with `clear` command

wmbrownIntel committed:
Fix to make the property list empty for command queues when timing disabled.

wmbrownIntel committed:
GPU Package fix where timing disable could result in event/marker destruction before completion on accelerator during initialization.

akohlmey committed:
more detailed log message with replicate

akohlmey committed:
more adjustments for bogus timer results on Intel OpenCL

akohlmey committed:
silence compiler warning

akohlmey committed:
preserve suffixp during clear command

akohlmey committed:
silence compiler warning

akohlmey committed:
simplify

akohlmey committed:
add -nonbuf flag to help output

sjplimp commented on src/replicate.cpp:
```diff
@@ -47,15 +47,16 @@ void Replicate::command(int narg, char **arg)
   int nrep = nx*ny*nz;
 
+  if (me == 0)
+    utils::logmesg(lmp,"Replicating atoms for a {}x{}x{} = {} times size system...\n", nx, ny, nz, nrep);
```
Replication is creating a {}x{}x{} = {}x larger system

akohlmey replied:
done

sjplimp reviewed (changes requested):
what did the TIP4P cutoff become ?

akohlmey committed:
cosmetic

akohlmey commented:
> what did the TIP4P cutoff become ?

Same as for the TIP4P GPU style: pair_style cutoff + q-distance + OH-bond distance + neighbor skin

sjplimp commented on src/replicate.cpp:
```diff
@@ -47,15 +47,16 @@ void Replicate::command(int narg, char **arg)
   int nrep = nx*ny*nz;
 
+  if (me == 0)
+    utils::logmesg(lmp,"Replication is creating a {}x{}x{} = {}x larger system...\n",nx,ny,nz,nrep);
```
sorry, though better of {}x larger, so instead:
= {} times larger system

akohlmey replied:
done as well

akohlmey committed:
check for and replace *really* old LAMMPS home page URL. apply fixes.

akohlmey committed:
correct implementation of quartic potential in pair style list

akohlmey committed:
add md5sums for plumed 2.7.5 and 2.8.1, update default version to 2.8.1

akohlmey committed:
try to make more portable (in case this ever gets ported to windows)

akohlmey committed:
find cythonize executable on recent FreeBSD versions

akohlmey committed:
Python support in ML-IAP requires NumPy. Check for it if CMake supports it.

akohlmey committed:
make Linux behavior default for loading Python shared lib

This adds portability to platforms like FreeBSD

akohlmey committed:
tweak for portability

akohlmey committed:
small sphinx tweaks. require sphinx 5.2 or later.

akohlmey committed:
add workaround for missing links to fortran functions in sphinx output

akohlmey committed:
replace one more suffix

akohlmey committed:
update Plumed download to version 2.8.1

akohlmey committed:
must bootstrap centos 7 from dockerhub now

akohlmey committed:
allow `-kokkos on g 0` without GPU support, report GPUs only if ngpu > 0

akohlmey committed:
update developer info

akohlmey committed:
silence unrecognized pragma warnings

akohlmey committed:
Merge remote-tracking branch 'github/develop' into collected-small-changes

# Conflicts:
#	src/DIPOLE/pair_lj_cut_dipole_long.h
#	src/OPENMP/fix_nvt_sllod_omp.h
#	src/PHONON/fix_phonon.h
#	src/SPIN/pair_spin_dipole_cut.h
#	src/SPIN/pair_spin_dipole_long.h
#	src/STUBS/mpi.h
#	src/UEF/fix_nh_uef.h
#	src/fix_nvt_sllod.h
#	tools/coding_standard/homepage.py

akohlmey committed:
add compatibility to VTK version 9.0 and above
